components/vars/tasks/updateDNA.js

```diff
@@ -0,0 +1,565 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const path = require('path');
+const fsp = require('fs').promises;
+const gulp = require('gulp');
+const ext = require('replace-ext');
+const logger = require('gulplog');
+const through = require('through2');
+
+const del = require('del');
+
+// Base variables we can just map directly
+let flatVars = [
+  'colorGlobals',
+  'fontGlobals',
+  'dimensionGlobals',
+  'animationGlobals',
+  'staticAliases'
+];
+
+let dropTokens = {
+  'name': true,
+  'description': true,
+  'status:': true,
+  'varBaseName': true
+};
+
+function stripReference(value) {
+  return value.replace(/(colorStopData|colorTokens|scaleData|dimensionTokens|colorAliases|dimensionAliases)\./g, '');
+}
+
+function getJSVariableReference(value) {
+  let reference = stripReference(value);
+  let parts = reference.split('.');
+  let finalReference = parts.shift() + parts.map(JSON.stringify).map(value => `[${value}]`).join('');
+  return finalReference;
+}
+
+function getExport(key, value) {
+  if (value[0] === '$') {
+    let reference = getJSVariableReference(value.substr(1));
+    return `exports[${JSON.stringify(key)}] = ${reference};\n`;
+  }
+  else {
+    return `exports[${JSON.stringify(key)}] = ${JSON.stringify(value)};\n`;
+  }
+}
+
+function getCSSVariableReference(value) {
+  if (value[0] === '$') {
+    value = value.substr(1);
+
+    // Strip the stop information
+    value = value.replace(/colorStopData\..*?\.colorTokens\./, 'global-color.');
+    value = value.replace(/colorStopData\..*?\.colorAliases\./, 'alias.');
+    value = value.replace(/colorStopData\..*?\.colorSemantics\./, 'semantic.');
+    // Strip the scale information
+    value = value.replace(/scaleData\..*?\.dimensionTokens\./, 'global-dimension.');
+    value = value.replace(/scaleData\..*?\.dimensionAliases\./, 'alias.');
+    // Sub in proper names for globals
+    value = value.replace(/colorGlobals\./, 'global-color.');
+    value = value.replace(/dimensionGlobals\./, 'global-dimension.static-');
+    value = value.replace(/fontGlobals\./, 'global-font.');
+    value = value.replace(/animationGlobals\./, 'global-animation.');
+    value = value.replace(/staticAliases\./, 'alias.');
```

Comment:
just an idea but should we maybe try to catch some errors here if any of these replaces go wrong?

Reply:
They'll either match or they won't; these operate on a single variable name.

We catch errors during build -- if this somehow references a non-existent variable, the build errors.